src/App.js

```diff
@@ -1,25 +1,99 @@
-import logo from './logo.svg';
-import './App.css';
+/** @jsxImportSource @emotion/react */
+import { css } from "@emotion/react";
+
+import { BREAK_POINT } from "./style/BreakPoints";
+
+import Header from "./components/Header";
+import BestProductsItems from "./components/BestProducts";
+import ProductsList from "./components/ProductsList";
+import { getBestProducts, getProducts } from "./API/api";
+import { useEffect, useState } from "react";
+import GlobalStyle from "./style/GlobalStyle";
+import Buttons from "./components/Buttons";
+import useViewportWidth from "./hooks/useViewportWidth";
 
 function App() {
+  const [products, setProducts] = useState([]);
+  const [bestProducts, setBestProducts] = useState([]);
+  const [bestPageSize, setBestPageSize] = useState(1);
+  const [page, setPage] = useState(1);
+  const [pageSize, setPageSize] = useState(4);
+  const [orderBy, setOrderBy] = useState("recent");
+  const [keyword, setKeyword] = useState("");
+  const [totalCount, setTotalCount] = useState();
+  const width = useViewportWidth();
+
+  useEffect(() => {
+    if (width < BREAK_POINT.md) {
+      setBestPageSize(1);
+      setPageSize(4);
+    } else if (width < BREAK_POINT.lg) {
+      setBestPageSize(2);
+      setPageSize(6);
+    } else {
+      setBestPageSize(4);
+      setPageSize(10);
+    }
+  }, [width]);
+
+  useEffect(() => {
+    //베스트 상품 fetch 함수
+    async function fetchBestProducts() {
+      try {
+        const result = await getBestProducts();
+        setBestProducts(result.list);
```

Comment:
```suggestion
        const { list } = await getBestProducts();
        setBestProducts(list);
```

이렇게 구조분해할당을 활용하시면 좀더 간소화되겠네요!

Reply:
그럼 밑에 전체상품 fetch 함수도 try...catch 문으로 수정해서 구조분해 할당으로 list와 totalCount를 받아와서 사용하는게 더 나은 방법일까요?

Reply:
ㅎㅎ 나중에 타입스크립트를 사용하게되면 조금 상황이 달라질거예요.

setBestProducts에 담기는 데이터의 타입이 undefined인 경우도 허용이 된다면 구조분해할당으로 바로 넣으셔도되고,
허용하고싶지않다면 s`etBestProducts(result?.list);` 로 유지해주시는게 좋습니다 :)